api/v1beta1/csiaddonsnode_types.go

```diff
@@ -0,0 +1,109 @@
+/*
+Copyright 2021.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+type CSIAddonsNodeDriver struct {
+	// This is the name of the CSI driver that this object refers to.
+	// This must be the same name returned by the CSI GetPluginame() call for
+	// that driver.
+	Name string `json:"name" protobuf:"bytes,1,opt,name=name"`
+
+	// Type represents the type of the driver i.e., rbd/cephfs.
+	Type string
```

Comment:
@nixpanic Can you please elaborate? It is present only in the CSIAddonsNodeDriver struct. And I think it will be good to have the `Type` to mention if it's the rbd/cephfs.  Not sure what exactly you meant here.

Reply:
consumers of the `CSIAddonsNode` do not need to know anything about the storage system. They would use the `drivername` to match `provisioner` in the StorageClass and get details to connect to the storage system that way.

A `Type` might be interesting for users (people) viewing the CR, but they would also see the `drivername`. I expect that users viewing the CR recognize the drivernames and know what storage system is used with it.